site/source/docs/porting/Debugging.rst

```diff
@@ -137,6 +137,58 @@ Debug printouts can even execute arbitrary JavaScript. For example::
   }
 
 
+Printing exception messages
+===========================
+
+Exceptions are thrown from WebAssembly using exception pointers, which means
+that try/catch/finally blocks in JavaScript will only receive a number, which
+represents a pointer into linear memory. In order to get the exception message,
+the user will need to create some WASM code which will extract the meaning from
+the exception. In the example code below we created a function that receives an
+``int`` which is pointer to an ``std::exception``, and by casting the pointer
+returns the ``what`` function call result.
+
+.. code-block:: cpp
+
+  #include <bind.h>
+
+  std::string getExceptionMessage(int exceptionPtr) {
+    return std::string(reinterpret_cast<std::exception *>(exceptionPtr)->what());
+  }
+
+  EMSCRIPTEN_BINDINGS(Bindings) {
+    emscripten::function("getExceptionMessage", &getExceptionMessage);
+  };
+
+Once such a function has been created, exception handling code in javascript
+can call it when receiving an exception from WASM. Here the function is used
+in order to log the thrown exception.
+
+.. code-block:: javascript
+
+  try {
+    ... // some code that calls WebAssembly
+  } catch (exception) {
+    console.error(Module.getExceptionMessage(exception));
+  } finally {
+    ...
+  }
+
+It's important to notice that this example code will work only for thrown
+statically allocated exceptions. If your code throws other objects, such as
+strings or dynamically allocated exceptions, the handling code will need to
+take that into account. For example, if your code throws either exception
+pointers or strings, the javascript code can handle these situations like
```

Comment:
`if your code throws either exception ￼pointers or strings`.. won't throwing strings will also end up calling `getExceptionMessage` since a C++ strings will look like i32 too.   Or do you mean a throw of a JS string from outside of wasm?

Reply:
No, I meant that if you throw something besides std::exception the example code can't handle it.

Reply:
I think this might make ti more clear:

`For example, if your code need to have both native C++ exceptions and JavaScript exceptions you could use the following code to distinguish between them:`

I was confused by the `or strings` part I think because I wasn't sure if you might have meant C++ strings.. but you did mean JS strings right?

Reply:
No. Since c++ allows you to throw a string, some people use strings as the standard thrown object, and then parse them for meaning in the catch scope.
But your explanation is also good, since I assume that throwing strings will only work from `embind`.